server/Makefile

```diff
@@ -0,0 +1,26 @@
+BUILD_DIR:=build
+NEPTUNE_SERVER_VERSION=v0.1.0
+BUILD_ERR_FILE=build-error.log
+ERR_DIR:=build-errors
+
+.PHONY: clean
+clean: ## clean build artifacts
+	rm -r ${BUILD_DIR} 2> /dev/null || :
+	rm -r ${ERR_DIR} 2> /dev/null || :
+
+.PHONY: build-all
+build-all: ## build for all targets
+	$(MAKE) clean
+	for target in aarch64-apple-darwin x86_64-apple-darwin x86_64-unknown-linux-gnu; do\
+		$(MAKE) build BUILD_TARGET=$$target; \
+	done
+
+.PHONY: build
+build: ## build release file
+	(ls ${BUILD_DIR} 2> /dev/null || mkdir ${BUILD_DIR})
+	(ls ${ERR_DIR} 2> /dev/null || mkdir ${ERR_DIR})
+	rustup target add ${BUILD_TARGET}
+	rustup update
```

Comment:
given that we're running this in a loop we shouldn't run it each time. Also here's the structure I think would make the most sense:
```
setup
# clean, rustup update, make build / error dirs if the don't exist

apple-m1
# rustup add aarch64-apple darwin
# compile etc

apple-intel
# rustup add aarch64-apple darwin
# compile etc

all:
# setup, apple-m1, apple-intel
```

make build is weird right now given we're expecting an argument to be passed (and this breaks the pattern of how make should be used), e.g. if I just run `make build` it fails. I'm curious if there are accepted patterns for this. if not, i would prefer just to replicate the code rather than break the standard make pattern by using env vars as arguments

Reply:
@ttarsi the main reason we have the build target in the first is for naming when cross-compiling. I found this crate `default-target`, which returns what we want in `BUILD_TARGET`. I think for now, since cross-compilation isn't working, we just have `make build` without args get the default target, and use that for naming and compilation.